server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -2065,10 +2071,14 @@ public boolean restartNetwork(Long networkId, boolean cleanup, boolean makeRedun
     public boolean restartNetwork(RestartNetworkCmd cmd) throws ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException {
         // This method restarts all network elements belonging to the network and re-applies all the rules
         Long networkId = cmd.getNetworkId();
+        NetworkVO network = getNetworkVO(cmd.getNetworkId(),"Nework to restart not found %s", String.valueOf(networkId));
         boolean cleanup = cmd.getCleanup();
+        if (network.getVpcId() != null && cleanup) {
```

Comment:
Wouldn't this break if restartVpc is called with cleanup=true (this has been historically there?)

Reply:
I have to check @rohityadavcloud . it is still work in progress (sorry, should hace created as draft)

Reply:
but my theory is that you would call `RestartVPC` and not `RestartNetwork` if you want a vpc router to be regenerated.